app/Http/Controllers/Crew/ScheduleController.php

```diff
@@ -12,8 +15,36 @@ class ScheduleController extends Controller
      * Returns the page with general management overview of the schedule
      * @return View
      */
-    public function index() : View
+    public function index(): View
     {
         return view('crew.schedule.index');
     }
+
+    /**
+     * Responsible for resetting the schedule based on the type
+     * $type == 'full' - removing all scheduling from the presentation; removing the timeslots and the opening/closing
+     * $type == 'scheduled' - removing all scheduling from the presentation
+     * @param $type
+     * @return mixed
+     */
+    public function reset($type)
+    {
+        Presentation::query()->update([
+            'room_id' => null,
+            'timeslot_id' => null,
+            'start' => null
+        ]);
+
+        if ($type == 'full') {
+            DefaultPresentation::truncate();
+
+            // Truncate does not work when model is FK
```

Comment:
As an relative outsider, I cannot find any documentation so I can understand this. It is not clear enough what the design of the "Schedule" is probably

Reply:
That would be addressed more carefully in the docs.